Dockerfile

```diff
@@ -18,7 +18,7 @@ COPY . /go/src/github.com/flyteorg/flytepropeller/
 
 # This 'linux_compile' target should compile binaries to the /artifacts directory
 # The main entrypoint should be compiled to /artifacts/flytepropeller
-RUN make linux_compile
+RUN go mod vendor && make linux_compile
```

Comment:
go mod download maybe?  we don't vendor anymore.  do you remember what the error was?  you shouldn't have to do this.

Reply:
This error only happens when I replace the idl version with another branch. like 
`replace github.com/flyteorg/flyteidl => github.com/flyteorg/flyteidl v0.21.13-0.20211206210958-804fa1685264
`
will remove this before we merge the PR

Reply:
Error message:
```bash
#15 0.319 	google.golang.org/grpc@v1.36.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#15 0.319 	google.golang.org/protobuf@v1.25.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#15 0.319 	k8s.io/api@v0.20.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#15 0.319 	k8s.io/apimachinery@v0.20.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#15 0.319 	k8s.io/client-go@v0.20.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#15 0.319 	k8s.io/klog@v1.0.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#15 0.319 	sigs.k8s.io/controller-runtime@v0.8.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#15 0.319 	github.com/aws/amazon-sagemaker-operator-for-k8s: is replaced in go.mod, but not marked as replaced in vendor/modules.txt
#15 0.319 	github.com/flyteorg/flyteidl: is replaced in go.mod, but not marked as replaced in vendor/modules.txt
#15 0.319 	github.com/flyteorg/flyteplugins: is replaced in go.mod, but not marked as replaced in vendor/modules.txt
#15 0.319 
#15 0.319 	To ignore the vendor directory, use -mod=readonly or -mod=mod.
#15 0.319 	To sync the vendor directory, run:
#15 0.319 		go mod vendor
#15 0.320 make: *** [Makefile:14: linux_compile] Error 1
------
executor failed running [/bin/sh -c make linux_compile]: exit code: 2
make: *** [docker_build] Error 1
```